### PR TITLE
Update netsurf.browser and netsurf.lib*, fix on darwin

### PR DESCRIPTION
**Pull request by felixalbrigtsen**
## Description of changes

Updates netsurf.browser from 3.10 to 3.11, and similar updates for all corresponding libraries(see commit messages).

- [changelog](https://download.netsurf-browser.org/netsurf/releases/ChangeLog.txt) for NetSurf/browser and various libs
- [changelog](https://www.netsurf-browser.org/projects/libcss/#changes-0.9.2) for libcss

These include a few minor but potentially breaking changes, like deprecating TLS 1.0 / 1.1.

Adds compile flags in browser.nix to allow building on darwin and with clang in general.

The build system is already updated in #277664.
Supersedes #277662 and #277669.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Remaining / TODO:
- [x] Test with uilib=framebuffer
  - Seems to work fine on NixOS, build fails on darwin
- [x] Test with uilib=gtk2
  - Builds fine, shows websites fine, but seems to be missing some UI icons (e.g. back button)
  - [ ] Fix missing resources
- [x] Test with uilib=sixel
  - Works fine on NixOS, but SDL_sixel is broken on darwin
- [ ] Enable tests in testPhase
- [ ] Enable/fix "Export to PDF" with libharu
- [x] Look into broken gimpPlugins, imv and toppler

---

Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.resynthesizer</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.gap</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gimp</li>
    <li>gimp.dev</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>netsurf.browser</li>
    <li>netsurf.libcss</li>
    <li>netsurf.libdom</li>
    <li>netsurf.libhubbub</li>
    <li>netsurf.libnsbmp</li>
    <li>netsurf.libnsgif</li>
    <li>netsurf.libnsutils</li>
    <li>netsurf.libparserutils</li>
    <li>netsurf.libsvgtiny</li>
  </ul>
</details>

<hr>

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.resynthesizer</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>imv</li>
    <li>imv.man</li>
    <li>toppler</li>
  </ul>
</details>
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gimp</li>
    <li>gimp-with-plugins</li>
    <li>gimp.dev</li>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gap</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>netsurf.browser</li>
    <li>netsurf.libcss</li>
    <li>netsurf.libdom</li>
    <li>netsurf.libhubbub</li>
    <li>netsurf.libnsbmp</li>
    <li>netsurf.libnsgif</li>
    <li>netsurf.libnsutils</li>
    <li>netsurf.libparserutils</li>
    <li>netsurf.libsvgtiny</li>
  </ul>
</details>

<hr>

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/browsers/netsurf/browser.nix [hunk 1]
```diff
@@ -103,8 +107,20 @@ stdenv.mkDerivation (finalAttrs: {
 
   env.NIX_CFLAGS_COMPILE = "-fcommon";
 
+  CFLAGS = lib.optionalString stdenv.isDarwin "-D_DARWIN_C_SOURCE";
```
Comment (AndersonTorres) [inline pkgs/applications/networking/browsers/netsurf/browser.nix:110]:
```suggestion
  env.CFLAGS = lib.optionalString stdenv.isDarwin "-D_DARWIN_C_SOURCE";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]
Please someone merge this!

**Comment by felixalbrigtsen** [comment]
I have not undrafted it yet as it still breaks `imv` and `toppler`, should that be ignored? I haven't quite figured out why they break yet. Also, the results from @wegank are a bit surprising as it builds fine on my aarch64-darwin machine.

**Comment by wegank** [comment]
According to [ofborg logs](https://logs.ofborg.org/?key=nixos/nixpkgs.285118&attempt_id=2c6a15b7-3a3b-4ca3-a0bf-62490cc5b1c6), `netsurf.libparserutils` might need `libiconv` as a dependency on x86_64-darwin.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 285118` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>gimp-with-plugins</li>
    <li>toppler</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gimp (gimpPlugins.gimp)</li>
    <li>gimp.dev (gimpPlugins.gimp.dev)</li>
    <li>gimpPlugins.gmic</li>
    <li>netsurf-browser (netsurf.browser)</li>
    <li>netsurf.libcss</li>
    <li>netsurf.libdom</li>
    <li>netsurf.libhubbub</li>
    <li>netsurf.libnsbmp</li>
    <li>netsurf.libnsgif</li>
    <li>netsurf.libnsutils</li>
    <li>netsurf.libparserutils</li>
    <li>netsurf.libsvgtiny</li>
  </ul>
</details>

**Comment by wegank** [comment]
Probably drop libnsgif from `withBackends` in imv? I don't see a patch for https://bugs.gentoo.org/922496 in other distros, most of them just pass `-Dlibnsgif=disabled`.

Also, toppler isn't a regression of this PR. It already fails on master.

**Comment by AndersonTorres** [comment]
> I have not undrafted it yet as it still breaks imv and toppler, should that be ignored?

This is not the first time we throw Darwin ~~under the bus~~ to the backlog.

**Comment by felixalbrigtsen** [comment]
Result of `nixpkgs-review pr 285118` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.resynthesizer</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>toppler</li>
  </ul>
</details>
<details>
  <summary>29 packages built:</summary>
  <ul>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gimp</li>
    <li>gimp-with-plugins</li>
    <li>gimp.dev</li>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gap</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>imv</li>
    <li>imv.man</li>
    <li>netsurf.browser</li>
    <li>netsurf.libcss</li>
    <li>netsurf.libdom</li>
    <li>netsurf.libhubbub</li>
    <li>netsurf.libnsbmp</li>
    <li>netsurf.libnsgif</li>
    <li>netsurf.libnsutils</li>
    <li>netsurf.libparserutils</li>
    <li>netsurf.libsvgtiny</li>
  </ul>
</details>

**Review by rnhmjoj — APPROVED** [review]
👍 for the imv part.

**Pull request merged**